planner/core/exhaust_physical_plans.go

```diff
@@ -1263,10 +1263,32 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 	return hashAggs
 }
 
+func (ps *PhysicalStreamAgg) convert2Enforced() {
+	for i := range ps.childrenReqProps {
+		ps.childrenReqProps[i].Enforced = true
+	}
+}
+
 func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProperty) []PhysicalPlan {
+	preferHash := (la.preferAggType & preferHashAgg) > 0
+	preferStream := (la.preferAggType & preferStreamAgg) > 0
+
+	hashAggs := la.getHashAggs(prop)
+	if hashAggs != nil && preferHash && !preferStream {
```

Comment:
Why do we need to check this here since we've checked this when building the logical plan?

Reply:
We're still discussing  whether to throw a error or just record a warning, so I still keep this check.

Reply:
I think we can throw a warning to specify that aggregate hints are conflicted.

Reply:
Okay, get it.